# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/RegExpShared.h
# Commit: 00d2e220a575
# Full Hash: 00d2e220a57535825b59e92df5201119109f55e4
# Author: Iain Ireland <iireland@mozilla.com>
# Date: 2020-05-22 21:38:53
# Regressor Bug: 1362154
# File Overlap Count: 1
# Description:
#   Bug 1362154: Part 5: Create groups property on result object r=mgaudet
#   
#   When a regexp has named captures, the result object has an extra `groups` property, which is an object that in turn has a string property per named capture. This patch creates the `groups` object and attaches it to the result object.
#   
#   The changes here correspond to this section of the spec proposal, which conveniently highlights the changes: https://tc39.es/proposal-regexp-named-groups/#sec-regexpbuiltinexec
# ==============================================================================

diff -r 1322042730b6 -r 00d2e220a575 js/src/vm/RegExpShared.h
--- a/js/src/vm/RegExpShared.h	Wed May 20 21:04:01 2020 +0000
+++ b/js/src/vm/RegExpShared.h	Wed May 20 21:04:47 2020 +0000
@@ -391,6 +391,9 @@
 
   static const size_t MatchResultObjectIndexSlot = 0;
   static const size_t MatchResultObjectInputSlot = 1;
+#ifdef ENABLE_NEW_REGEXP
+  static const size_t MatchResultObjectGroupsSlot = 2;
+#endif
 
   /* Get or create template object used to base the result of .exec() on. */
   ArrayObject* getOrCreateMatchResultTemplateObject(JSContext* cx) {
